### PR TITLE
Add example for a stacked horizontal bar chart

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

There's only one example for a simple (vertical) stacked bar chart. I think this is a worthwhile addition:

- It's showing the general approach how to stack multiple bars.
- It's illustrates a certain kind of common diagram.
- It shows how to draw labels with values at the bar centers.

[Example:](https://21315-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/statistics/horizontal_barchart_distribution.html#sphx-glr-gallery-statistics-horizontal-barchart-distribution-py)

image

### CODE REVIEW

File: examples/statistics/horizontal_barchart_distribution.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#
+# References
+# """"""""""
+#
+# The use of the following functions, methods, classes and modules is shown
+# in this example:
+
+import matplotlib
+matplotlib.axes.Axes.barh
+matplotlib.pyplot.barh
```
Comment (efiring) [inline examples/statistics/horizontal_barchart_distribution.py]: Why are the pyplot versions included in this list, when they aren't used in the example?

Comment (ImportanceOfBeingErnest) [reply examples/statistics/horizontal_barchart_distribution.py]: This reference section 

* provides a link to the function doc page
* ensures that the function doc page contains a link to the respective example

For the first point, one could argue that no such link is needed. The second point however makes this useful; people often arrive at the pyplot function docs (e.g. by googling "matplotlib barh") - in this case they might want to see the same example linked, even if it does not necessarily use `plt.barh`, but `ax.barh` instead.

**Comment by ImportanceOfBeingErnest** [comment]
I like the example (and its very clean), so approving. I do wonder though if it would not be better put in the same section as other `bar` examples, because the functionality that is presented is not limited to statistical representations.

**Comment by timhoffm** [comment]
Fair point I’ll move the example to the other section.

**Comment by timhoffm** [comment]
Moved to section "lines, bars and markers". This is ready to go.

**Pull request merged**